### PR TITLE
[is-valid-glob] Bugfix: Function can accept no arguments

### PR DESCRIPTION
**Pull request by joshuaharry**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
See https://github.com/micromatch/is-valid-glob/blob/c18c82999df11133c5946f9a21ef74a0ebf88334/test.js#L18 and https://github.com/micromatch/is-valid-glob/blame/master/README.md#L37

### CODE REVIEW

File: types/is-valid-glob/index.d.ts [hunk 1]
```diff
@@ -5,4 +5,4 @@
 
 export = isValidGlob;
 
-declare function isValidGlob(input: any): input is string | string[];
+declare function isValidGlob(input?: any): input is string | string[];
```
Comment (joshuajeschek) [inline types/is-valid-glob/index.d.ts:8]:
```suggestion
declare function isValidGlob(input: any): input is string | string[];
declare function isValidGlob(): false;
```

File: types/is-valid-glob/is-valid-glob-tests.ts [hunk 2]
```diff
@@ -5,3 +5,5 @@ const input: any = 'foo/*.js';
 if (isValidGlob(input)) {
     input; // $ExpectType string | string[]
 }
+
+isValidGlob(); // $ExpectType boolean
```
Comment (joshuajeschek) [inline types/is-valid-glob/is-valid-glob-tests.ts:9]:
```suggestion
isValidGlob(); // $ExpectType false
```

**Review by joshuajeschek — CHANGES_REQUESTED** [review]
What do you think about this? This way, users are informed that this is not the intended way by getting informed that this will always evaluate to false!

**Comment by joshuaharry** [comment]
Great suggestions, thanks!

**Review by joshuajeschek — APPROVED** [review]
Looks good now!

**Comment by joshuaharry** [comment]
Ready to merge

**Pull request merged**